linera-execution/src/wasm/runtime_actor/requests.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Different request types for different runtimes.
+
+use super::responses::{SyncOrAsyncResponse, SyncResponseSender};
+use crate::{CallResult, UserApplicationId};
+use futures::channel::oneshot;
+use linera_base::{
+    data_types::{Amount, Timestamp},
+    identifiers::{ChainId, SessionId},
+};
+use linera_views::batch::Batch;
+use std::fmt::{self, Debug, Formatter};
+
+/// Requests shared by contracts and services.
+pub enum BaseRequest {
```

Comment:
Seems like there could be an opportunity to extend `linera-witty(-derive)` to support actors?

Reply:
Yep! I previously did something similar as a [separate crate prototype](https://github.com/jvff/ezrpc).